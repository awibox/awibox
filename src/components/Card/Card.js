import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
// Styles
import styles from './Card.scss';

const Card = (props) => (
  <div className={classNames(styles.card, {
    [props.className]: props.className,
    [styles[props.color]]: props.color,
  })}>
    {props.title && <div className={styles.title}>
      {props.icon && <i className={`fa fa-${props.icon}`} />}
      {props.title}
    </div>}
    {!props.withoutContainer
      ? <div className={styles.content}>{props.children}</div>
      : props.children
    }
  </div>
);

Card.propTypes = {
  children: PropTypes.any.isRequired,
  className: PropTypes.string,
  icon: PropTypes.string,
  withoutContainer: PropTypes.bool,
  title: PropTypes.string,
  color: PropTypes.string,
};

export default Card;
