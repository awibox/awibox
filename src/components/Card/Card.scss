@import '../../styles/variables';

.card {
  padding: $cardIndent;
  border-top: 3px solid rgba($darkGray, .5);
  background: #fff;
  box-shadow: 0 1px 3px rgba(0, 0, 0, 0.12), 0 1px 2px rgba(0, 0, 0, 0.24);
  box-sizing: border-box;
  transition: all 0.3s cubic-bezier(.25, .8, .25, 1);
  border-radius: 3px;
  margin-bottom: $cardIndent;
  &.green {
    border-color: $green;
  }
  &.blue {
    border-color: $blue;
  }
  &.orange {
    border-color: $orange;
  }
}
.title {
  border-bottom: 2px solid #e7eaec;
  width: 100%;
  margin-left: -$cardIndent;
  padding: 0 $cardIndent $cardIndent;
  margin-bottom: $cardIndent;
  font-size: 14px;
  height: 14px;
  font-weight: 600;
}

