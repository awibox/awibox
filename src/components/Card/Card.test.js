import React from 'react';
import { shallow } from 'enzyme';
import Card from './Card';

const testClassName = 'test-class';

describe('Card', () => {
  it('should render correctly', () => {
    const component = shallow(<Card
      title={'Title'}
      icon={'fa fa-home'}
      color={'blue'}
      className={testClassName}
      withoutContainer={false}>
      <div>
        <div>
          Test Card
        </div>
      </div>
    </Card>);
    expect(component).toMatchSnapshot();
  });
});
