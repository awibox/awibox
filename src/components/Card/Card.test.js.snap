// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render correctly 1`] = `
<div
  className="card"
>
  <div
    className="content"
  >
    <div>
      <div>
        Test Card
      </div>
    </div>
  </div>
</div>
`;
